### PR TITLE
Combine `_LocalApp` and `_ContainerApp` and move them to `modal.running_app`

### PR DESCRIPTION
**Pull request by erikbern**
1. Combine the two (very simple dataclasses) into one.
2. Move them to a new module.

`modal.app` is basically available now, which will let us rename `modal.stub` to `modal.app` 🎉

### CODE REVIEW

**Comment by erikbern** [comment]
Btw is there a way to remove `app.py` in the same PR as I rename `stub.py` to `app.py`? I'm a bit nervous the git history will be lost of we squash the commits (since git might thing `stub.py` was removed and the same code added in `app.py` unless it's a "clean" file rename)

**Comment by mwaskom** [comment]
> Btw is there a way to remove app.py in the same PR as I rename stub.py to app.py? I'm a bit nervous the git history will be lost of we squash the commits (since git might think stub.py was removed and the same code added in app.py unless it's a "clean" file rename). Looks like we don't support other merging strategies than squashing.

We can temporarily disable the squash requirement, I've done that in the past to support other housekeeping where we wanted to preserve history.

File: modal/running_app.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright Modal Labs 2024
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from google.protobuf.message import Message
+
+
+@dataclass
+class RunningApp:
```
Comment (freider) [inline modal/running_app.py:9]: Full circle! Didn't we use to have something called `RunningApp`, that was the context var of the stub.run context manager (which was then `app.run()` iirc) ? :D

**Review by freider — APPROVED** [review]
Clean 💯

File: modal/_container_entrypoint.py [hunk 2]
```diff
@@ -490,10 +490,12 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
                 client,
             )
 
+        # Get ids and metadata for objects (primarily functions and classes) on the app
+        container_app: RunningApp = container_io_manager.get_app_objects()
```
Comment (mwaskom) [inline modal/_container_entrypoint.py:494]: Any functional consequences of moving this out of the if block below?

**Review by mwaskom — APPROVED** [review]
`modal.app` is dead; long live `modal.app`!

`RunningApp` is kind of a clumsy name IMO given that there's no application logic here anymore. Maybe `AppInfo` or something? That's lame too, though, so not a final suggestion :)

File: modal/_container_entrypoint.py (hunk 2 above)
Comment (erikbern) [reply modal/_container_entrypoint.py:494]: I noticed we're actually referring to it in the higher-level scope, in the next block. We basically had code like this

```python
if a:
    x = X()

if b:
    x.f()
```

I _think_ that the 2nd block is never entered unless the 1st block is also entered, so I don't think there's a practical difference. I'm surprised mypy or ruff doesn't catch issues like this actually.

**Comment by erikbern** [comment]
> `RunningApp` is kind of a clumsy name IMO given that there's no application logic here anymore. Maybe `AppInfo` or something? That's lame too, though, so not a final suggestion :)

Yeah I don't love it. It does specifically refer to a running app though, whereas I don't think `AppInfo` makes that clear.

File: modal/running_app.py (hunk 1 above)
Comment (erikbern) [reply modal/running_app.py:9]: I think that was basically the `App` class, which I split up at some point, then eventually merged back (after deleting like 99% of the code however)

**Pull request merged**